src/morphing.rs

```diff
@@ -37,6 +41,130 @@ pub extern fn morph_object(object: &[u8], request: &str) -> *const u8 {
 /// # Arguments
 ///
 /// `html` - HTML page.
-fn morph_html(html: &mut Object) -> usize {
+fn morph_html(html: &mut Object) -> Result<usize, ()> {
+    let mut objects = parse_objects(html);
+    objects.sort_unstable_by(|ref o1, ref o2|
+                                o1.content.len().cmp(&o2.content.len()));
+    // Minimum characteristics.
+    let min_count = objects.len();
+    let html_min_size = html.content.len();
+    let obj_min_size = objects.iter()
+                              .map(|o| o.content.len())
+                              .min()
+                              // TODO: what should we do in this case?
```

Comment:
I don't think we should be defining the `obj_min_size` var in the first place. By ensuring all `target_sizes` are greater than the smallest object on the page in `morph_from_distribution` we hurt the anonymity of the defense because we're only sampling from a fraction of the size distribution.